src/Service/AvatarService.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace App\Service;
+
+use Exception;
+use Symfony\Component\HttpFoundation\File\Exception\FileException;
+use Symfony\Component\HttpFoundation\File\UploadedFile;
+use Symfony\Component\String\Slugger\SluggerInterface;
+
+class AvatarService
+{
+    private string $targetDirectory;
+    private SluggerInterface $slugger;
+
+    public function __construct($targetDirectory, SluggerInterface $slugger)
+    {
+        $this->targetDirectory = $targetDirectory;
+        $this->slugger = $slugger;
+    }
+
+    /**
+     * @throws Exception
+     */
+    public function upload(UploadedFile $file): string
+    {
+        $originalFilename = pathinfo($file->getClientOriginalName(), PATHINFO_FILENAME);
+        $safeFilename = $this->slugger->slug($originalFilename);
+        $fileName = $safeFilename.'-'.uniqid().'.'.$file->guessExtension();
+
+        try {
+            $file->move($this->getTargetDirectory(), $fileName);
```

Comment:
je ne suis pas sûr qu'il soit nécessaire d'ajouter un getter ici, je pense que tu peux remplacer par $this->targetDirectory et supprimer le getter. essaie pour voir

Reply:
Effectivement, le getter n'est pas forcément nécessaire. J'ai mis à jour, testé et tout fonctionne parfaitement.